### PR TITLE
STYLE enable pylint: confusing-with-statement

### PR DESCRIPTION
**Pull request by natmokval**
Issue #48855. This PR enables pylint type "W" warning: `confusing-with-statement`
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

After fixing the warning `confusing-with-statement` running `pandas/tests/io/test_pickle.py` raises an error `EOFError: Ran out of input`. To handle this error I check if the size of the file is not zero. Maybe there is a better way to deal with the test?

### CODE REVIEW

File: pandas/tests/io/test_pickle.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def compress_file(self, src_path, dest_path, compression):
             raise ValueError(msg)
 
         if compression not in ["zip", "tar"]:
-            with open(src_path, "rb") as fh, f:
+            with open(src_path, "rb") as fh:
                 f.write(fh.read())
```
Comment (MarcoGorelli) [inline pandas/tests/io/test_pickle.py:287]: does it work to nest the with statements? like
```python
with open(src_path, 'rb') as fh:
    with f:
        f.write(fh.read())
```
?

Comment (natmokval) [reply pandas/tests/io/test_pickle.py]: Thanks, @MarcoGorelli, it works like a charm. I updated the commit.

File: pandas/tests/io/test_pickle.py [hunk 2]
```diff
@@ -352,7 +353,6 @@ def test_read_explicit(self, compression, get_random_path):
 
             # read compressed file
             df2 = pd.read_pickle(p2, compression=compression)
-
```
Comment (MarcoGorelli) [inline pandas/tests/io/test_pickle.py:355]: for the future, it's better to not delete unrelated lines, but it's fine here

**Review by MarcoGorelli — APPROVED** [review]
thanks @natmokval !

**Pull request merged**